jerry-ext/arg/arg-transform-functions.c

```diff
@@ -84,14 +87,120 @@ jerryx_arg_transform_number (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
+} /* jerryx_arg_transform_number_strict */
+
+/**
+ * Tranform a JS argument to a double. Type coercion is allowed.
+ *
+ * @return jerry undefined: the transformer passes,
+ *         jerry error: the transformer fails.
+ */
+jerry_value_t
+jerryx_arg_transform_number (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< available JS args */
+                             const jerryx_arg_t *c_arg_p) /**< the native arg */
+{
+  return jerryx_arg_transform_number_common (js_arg_iter_p, c_arg_p->dest);
 } /* jerryx_arg_transform_number */
 
 /**
+ * Helper function to process a double number before converting it
+ * to an integer.
+ *
+ * @return jerry undefined: the transformer passes,
+ *         jerry error: the transformer fails.
+ */
+static jerry_value_t
+jerryx_arg_helper_process_double (double *d, /**< [in, out] the number to be processed */
+                                  double min, /**< the min value for clamping */
+                                  double max, /**< the max value for clamping */
+                                  jerryx_arg_int_option_t option) /**< the converting policies */
+{
+  if (option.clamp == JERRYX_ARG_NO_CLAMP)
+  {
+    if (*d > max || *d < min)
+    {
+      return jerry_create_error (JERRY_ERROR_TYPE,
+                                 (jerry_char_t *) "The number is out of range.");
+    }
+  }
+  else
+  {
+    *d = *d < min ? min : *d;
+    *d = *d > max ? max : *d;
+  }
+
+  if (option.round == JERRYX_ARG_ROUND)
+  {
+    *d = (*d >= 0.0) ? floor (*d + 0.5) : ceil (*d - 0.5);
+  }
+  else if (option.round == JERRYX_ARG_FLOOR)
+  {
+    *d = floor (*d);
+  }
+  else
+  {
+    *d = ceil (*d);
+  }
+
+  return jerry_create_undefined ();
+} /* jerryx_arg_helper_process_double */
+
+/**
+ * Use the macro to define two transform functions for int type.
+ * One is non-strict version and the other is the strict version.
+ */
+#define JERRYX_ARG_TRANSFORM_FUNC_FOR_INT_TEMPLATE(type, suffix, min, max) \
+  jerry_value_t jerryx_arg_transform_ ## type ## suffix (jerryx_arg_js_iterator_t *js_arg_iter_p, \
+                                                          const jerryx_arg_t *c_arg_p) \
+  { \
+    double tmp; \
+    jerry_value_t rv = jerryx_arg_transform_number ## suffix ## _common (js_arg_iter_p, &tmp); \
+    if (jerry_value_has_error_flag (rv)) \
+    { \
+      return rv; \
+    } \
+    jerry_release_value (rv); \
+    jerryx_arg_int_option_t *options_p = (jerryx_arg_int_option_t *) &c_arg_p->extra_info; \
+    rv = jerryx_arg_helper_process_double (&tmp, min, max, *options_p); \
+    if (jerry_value_has_error_flag (rv)) \
+    { \
+      return rv; \
+    } \
+    *(type ## _t *) c_arg_p->dest = (type ## _t) tmp; \
+    return rv; \
+  } \
+
+#define JERRYX_ARG_TRANSFORM_FUNC_FOR_INT(type, min, max) \
+  JERRYX_ARG_TRANSFORM_FUNC_FOR_INT_TEMPLATE(type, _strict, min, max) \
+  JERRYX_ARG_TRANSFORM_FUNC_FOR_INT_TEMPLATE(type, , min, max)
```

Comment:
missing space at `TEMPLATE(` (between end of macro name and opening parenthesis)